compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1190,8 +1231,35 @@ public void visit(BLangUserDefinedType userDefinedTypeNode) {
         //    If the package alias is not empty or null, then find the package scope,
         if (symbol == symTable.notFoundSymbol) {
             BSymbol tempSymbol = lookupMainSpaceSymbolInPackage(userDefinedTypeNode.pos, env, pkgAlias, typeName);
+
             if ((tempSymbol.tag & SymTag.TYPE) == SymTag.TYPE) {
                 symbol = tempSymbol;
+            } else if (Symbols.isSymTagOn(tempSymbol, SymTag.VARIABLE) && env.node.getKind() == NodeKind.FUNCTION) {
+                BLangFunction func = (BLangFunction) env.node;
+
+                if (func.returnTypeNode != null && func.hasBody() &&
+                        func.body.getKind() == NodeKind.EXTERN_FUNCTION_BODY) {
+                    BType paramValType = getTypedescParamValueType(func.requiredParams, tempSymbol,
+                                                                   func.returnTypeNode.pos);
+
+                    if (paramValType == symTable.semanticError) {
+                        this.resultType = symTable.semanticError;
+                        return;
+                    }
+
+                    BTypeSymbol tSymbol = new BTypeSymbol(SymTag.TYPE, Flags.PARAMETERIZED | tempSymbol.flags,
+                                                          tempSymbol.name, tempSymbol.pkgID, null, func.symbol);
+                    tSymbol.type = new BParameterizedType(paramValType, (BVarSymbol) tempSymbol,
+                                                          tSymbol, tempSymbol.name);
+
+                    this.resultType = tSymbol.type;
+                    this.resultType.flags |= Flags.PARAMETERIZED;
+                    return;
+                } else {
+                    dlog.error(userDefinedTypeNode.pos, DiagnosticCode.INVALID_USE_OF_TYPEDESC_PARAM);
```

Comment:
@MaryamZi Should I do the same check done a few lines below this (L1272) here as well when logging errors for user defined types?